kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/SectionEditor.vue

```diff
@@ -1,21 +1,407 @@
+            :text="applySettings$()"
+            class="apply-settings-style"
+            @click="applySettings"
+          />
+
+        </KGridItem>
+      </KGrid>
+    </div>
   </div>
 
 </template>
 
 
 <script>
 
+  import { enhancedQuizManagementStrings } from 'kolibri-common/strings/enhancedQuizManagementStrings';
+  import useKResponsiveWindow from 'kolibri.coreVue.composables.useKResponsiveWindow';
+  import Draggable from 'kolibri.coreVue.components.Draggable';
+  import DragContainer from 'kolibri.coreVue.components.DragContainer';
+  import DragHandle from 'kolibri.coreVue.components.DragHandle';
+
   export default {
     name: 'SectionEditor',
+    components: {
+      Draggable,
+      DragContainer,
+      DragHandle,
+    },
+    inject: ['quizForge'],
+    setup() {
+      const {
+        sectionSettings$,
+        sectionTitle$,
+        numberOfQuestionsLabel$,
+        optionalDescriptionLabel$,
+        quizResourceSelection$,
+        numberOfSelectedResources$,
+        currentSection$,
+        deleteSectionLabel$,
+        applySettings$,
+        changeResources$,
+        sectionOrder$,
+        questionOrder$,
+        randomizedLabel$,
+        randomizedOptionDescription$,
+        fixedLabel$,
+        fixedOptionDescription$,
+      } = enhancedQuizManagementStrings;
+      const { windowIsLarge, windowIsSmall } = useKResponsiveWindow();
+      return {
+        windowIsLarge,
+        windowIsSmall,
+        sectionSettings$,
+        sectionTitle$,
+        numberOfQuestionsLabel$,
+        optionalDescriptionLabel$,
+        quizResourceSelection$,
+        numberOfSelectedResources$,
+        currentSection$,
+        deleteSectionLabel$,
+        applySettings$,
+        changeResources$,
+        sectionOrder$,
+        questionOrder$,
+        randomizedLabel$,
+        randomizedOptionDescription$,
+        fixedLabel$,
+        fixedOptionDescription$,
+      };
+    },
+    data() {
+      return {
+        selectedQuestionOrder: true,
+        numQuestions: 1,
+        descriptionLabel: '',
+        sectionTitle: '',
+      };
+    },
+    computed: {
+      borderStyle() {
+        return `border: 1px solid ${this.$themeTokens.fineLine}`;
+      },
+      activeSection() {
+        return {
+          backgroundColor: this.$themePalette.grey.v_50,
+          border: `1px solid ${this.$themeTokens.fineLine}`,
+        };
+      },
+      dividerStyle() {
+        return `color : ${this.$themeTokens.fineLine}`;
+      },
+      /**
+       * @returns { QuizSection[] }
+       */
+      sectionOrderList() {
+        return this.quizForge.allSections.value;
+      },
+    },
+    methods: {
+      handleSectionSort(e) {
+        this.quizForge.updateQuiz({ question_sources: e.newArray });
+      },
+      applySettings() {
+        this.quizForge.updateSection({
+          section_id: this.quizForge.activeSection.value.section_id,
+          section_title: this.sectionTitle,
```

Comment:
The final stretch here will be wiring up the radio buttons for the randomization, question count, and description and this should be all set here

Reply:
A related issue is that the section's current settings aren't being applied to the form when it is initialized. For example, the first default created section has the title "Section 1" which should populate the "Title" textfield.